src/tools/synthetic_node.rs

```diff
@@ -138,6 +146,21 @@ impl SyntheticNode {
         }
     }
 
+    /// Attempts to read a message from the inbound (internal) queue of the node before the
+    /// timeout duration has elapsed.
+    pub async fn recv_message_timeout(
+        &mut self,
+        duration: Duration,
+    ) -> io::Result<(SocketAddr, Payload)> {
+        match timeout(duration, self.recv_message()).await {
+            Ok(message) => Ok(message),
```

Comment:
Can we just map err here and let Ok through?

Reply:
hm, let me try it 🕵️‍♀️

Reply:
Nope. I tried. But I couldn't make it work. If you have any suggestion which works, please share; I can update it in a separate commit